File: Algorithms/Divide and conquer/Quick Sort/python/quicksort.py
# Python program for quick sort
def partition(array,low,high):
    pivot = array[high]
    i = low -1
    for j in range(low,high):
        if array[j] < pivot:
            i=i+1
            # swap array[i] & array[j]
            (array[i],array[j]) = (array[j],array[i])
    # swap array[i+1] & array[high]
    (array[i+1],array[high])=(array[high],array[i+1])
    return i +1
def quickSort(array,low,high):
    if low < high:
        pi = partition(array,low,high)
        quickSort(array,low,pi-1)
        quickSort(array,pi+1,high)

data = [3, 11, 23, 6, 1, 15]
size = len(data)
quickSort(data,0,size -1)
print(data)

File: Algorithms/Divide and conquer/Quick Sort/c++/quickSort.cpp
/* C++ program for quick Sort*/
#include <iostream>

using namespace std;

void swap(int &a,int &b)
{
    int temp = a;
    a = b;
    b = temp;
}

int parti(int arr[],int low,int high)
{
    // int pivot = arr[high];
    int i = low -1;
    for(int j= low; j<high; j++) {
        if(arr[j]<=arr[high]) {
            i++;
            swap(arr[i],arr[j]);

        }
    }
    swap(arr[i+1],arr[high]);
    return i+1;
}

void quicksort(int arr[],int low, int high)
{
    if(low< high) {
        int pi = parti(arr,low,high);

        quicksort(arr,low,pi -1);
        quicksort(arr,pi+1,high);
    }
}

void printArray(int arr[],int size)
{
    for(int i = 0; i<size; i++) {
        cout << arr[i] <<" ";

    }
    cout << endl;
}

int main()
{
    int data[] = {3, 11, 23, 6, 1, 15};
    int size = sizeof(data)/sizeof(data[0]);
    quicksort(data,0,size -1);
    printArray(data,size);
}
